test-harness/infratests/unit.go

```diff
@@ -42,19 +40,19 @@ type UnitTestFixture struct {
 func RunUnitTests(fixture *UnitTestFixture) {
 	terraform.Init(fixture.GoTest, fixture.TfOptions)
 
-	workspace_name := random.UniqueId()
-	terraform.WorkspaceSelectOrNew(fixture.GoTest, fixture.TfOptions, workspace_name)
-	defer terraform.RunTerraformCommand(fixture.GoTest, fixture.TfOptions, "workspace", "delete", workspace_name)
+	workspaceName := "unit-tst-" + random.UniqueId()
+	terraform.WorkspaceSelectOrNew(fixture.GoTest, fixture.TfOptions, workspaceName)
```

Comment:
if we're creating one workspace for each one of our parallel test runs, then wouldn't that conflict with multiple calls to `RunUnitTests`?

Reply:
Good call. I suppose we want to limit the max number of terraform-based tests to 1 per host per working directory. Does this match your thinking?

Reply:
we could do that or support ♾️ test runs per working directory and use the same terraform workspace. I don't know the shortcomings of workspace's well enough to fully understand the risks with this approach. The other alternative is only supporting sequential test runs per working directory.

Reply:
My understanding is that terraform operations over the same workspace (i.e., state file) should be limited

https://learn.hashicorp.com/terraform/development/running-terraform-in-automation
> It is strongly recommended to use a backend that supports remote state, since that allows Terraform to automatically save the state in a persistent location where it can be found and updated by subsequent runs. Selecting a backend that supports state locking will additionally provide safety against race conditions that can be caused by concurrent Terraform runs.

Reply:
I suppose we can consider running each test deployment under the following condition:
 - Isolated temp directory: this lets us use a different workspace for each test. The state for this is kept, i believe, in `.terraform/` so we want a unique directory per test.
 - Unique workspace name (already done): this lets us run multiple deployments at once

Reply:
In any case, I'd like to go ahead and punt this to a different user story as it is outside the scope of integration testing only.

I've created https://github.com/microsoft/cobalt/issues/141 to track this work